### PR TITLE
fix: bid table reload fix & improve handle actions

### PR DESCRIPTION
**Pull request by domw30**
[Associated Notion Card](https://www.notion.so/zerotech/0-17-0-zAuction-2-1-network-switcher-a2c4a3409df541a197682d740babc572?v=c4f4a1df78154a0d9cb575d1fac426c5&p=9f09e355f71d4f67bcd609760be3202b)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
- [x] Bug
- [x] Improvements

## 3. What is the old behaviour?
https://user-images.githubusercontent.com/39112648/180245497-b428e29f-5325-4836-9cdf-70432384e7d9.mov

Data refetch is triggered due to state when selecting one of the options.
If the user owns the domain, they are able to 'Rebid' on the domain.

## 4. What is the new behaviour?
https://user-images.githubusercontent.com/39112648/180245847-dc8132ee-185a-49a4-b4e9-533ce0f4dde4.mov

Reload no longer occurs when selecting an option.
'Rebid' will not be an option if the user owns the domain.

### CODE REVIEW

File: src/containers/Tables/BidTable/BidTable.tsx [hunk 1]
```diff
@@ -1,111 +1,43 @@
...
-						openMakeBid={openMakeBid}
-						openCancelBid={openCancelBid}
-					/>
+					<BidTableRow {...props} refetch={refetch} />
 				)}
 				gridComponent={(props: any) => (
-					<BidTableCard
-						{...props}
-						onRefetch={refetch}
-						openMakeBid={openMakeBid}
```
Comment (padala) [inline src/containers/Tables/BidTable/BidTable.tsx:102]: @domw30  any specific reason we moved them down to the components. As it results in code duplication too

Comment (domw30) [reply src/containers/Tables/BidTable/BidTable.tsx:102]: The state higher up was causing the re-render. The changes here mirrored Bretts fix for the same issue on SubDomain Table. Strange one, but handling state higher up for modal and selected bid was causing each row/card to refetch data on every click event.

Comment (padala) [reply src/containers/Tables/BidTable/BidTable.tsx:102]: that's indeed strange need further investigation

File: src/containers/Tables/BidTable/BidTableRow/BidTableRow.tsx [hunk 2]
```diff
@@ -1,67 +1,120 @@
...
+						onClose={closeModal}
+						paymentTokenInfo={selectedBid.paymentTokenInfo}
+					/>
+				</Overlay>
+			)}
+			<tr className={styles.Container} data-testid={TestId.ROW_CONTAINER}>
+				<td>
+					<Artwork
+						data-testid={TestId.ARTWORK}
+						domain={'0://' + bid.domainName}
```
Comment (padala) [inline src/containers/Tables/BidTable/BidTableRow/BidTableRow.tsx:88]: assuming this is fine in zero.live too

File: src/containers/Tables/BidTable/BidTableRow/BidTableRow.tsx [hunk 3]
```diff
@@ -1,67 +1,120 @@
...
+	const closeModal = () => {
+		setModal(undefined);
+		setSelectedBid(undefined);
+	};
+
 	const onSelectOption = (option: Option) => {
 		switch (option.title) {
 			case ActionKeys.REBID:
-				props.openMakeBid?.(bid);
+				openMakeBid?.(bid);
```
Comment (padala) [inline src/containers/Tables/BidTable/BidTableRow/BidTableRow.tsx:55]: We can remove `?.` here

File: src/containers/Tables/BidTable/BidTableCard/BidTableCard.tsx [hunk 4]
```diff
@@ -1,86 +1,135 @@
...
+
 	const onSelectOption = (option: Option) => {
 		switch (option.title) {
 			case ActionKeys.REBID:
-				props.openMakeBid?.(bid);
+				openMakeBid?.(bid);
 				return;
 			case ActionKeys.CANCEL_BID:
-				props.openCancelBid?.(bid);
+				openCancelBid?.(bid);
```
Comment (padala) [inline src/containers/Tables/BidTable/BidTableCard/BidTableCard.tsx:62]: Can remove `?.` here

File: src/containers/Tables/BidTable/BidTableRow/BidTableRow.tsx (hunk 2 above)
Comment (domw30) [reply src/containers/Tables/BidTable/BidTableRow/BidTableRow.tsx:88]: Yeah just double checked with network switching and all looks good

File: src/containers/flows/CancelBid/components/Details.tsx [hunk 5]
```diff
@@ -16,7 +17,7 @@ const Details = ({
 	onNext,
 	paymentTokenInfo,
 }: DetailsProps) => (
-	<>
+	<div className={styles.NFTDetailsContainer}>
```
Comment (domw30) [inline src/containers/flows/CancelBid/components/Details.tsx:20]: Same style as MakeABid

BEFORE
Screenshot 2022-07-22 at 12 27 12

AFTER
Screenshot 2022-07-22 at 12 27 03


BEFORE
Screenshot 2022-07-22 at 12 27 40

AFTER
Screenshot 2022-07-22 at 12 26 51

**Review by padala — APPROVED** [review]

**Pull request merged**